building_dialouge_webapp/heat/views.py

```diff
@@ -228,9 +230,136 @@ class Results(SidebarNavigationMixin, TemplateView):
+        })
+
+        # Tabellen für Szenario 2
+        df_scenario2_investments = create_table("Investitionskosten", scenario2_dynamic_column_investments, scenario2_investments)
+        df_scenario2_subsidies = create_table("Zuschüsse", scenario2_dynamic_column_subsidies, scenario2_subsidies)
+        df_scenario2_sum = pd.DataFrame({
+            "Summe": [""],
+            "": [f"{sum(scenario2_investments) - sum(scenario2_subsidies):,.2f} €"]
+        })
+
+        # HTML-Tabellen generieren
+        def generate_html_table(investments, subsidies, sums, scenario_name):
+            investments_html = investments.to_html(classes=f"table {scenario_name}", index=False, escape=False)
+            subsidies_html = subsidies.to_html(classes=f"table {scenario_name}", index=False, escape=False)
+            sums_html = sums.to_html(classes=f"table {scenario_name}", index=False, escape=False)
+
+            return investments_html + subsidies_html + sums_html
+
+        html_scenario1 = generate_html_table(df_scenario1_investments, df_scenario1_subsidies, df_scenario1_sum, "tab_scenario1")
+        html_scenario2 = generate_html_table(df_scenario2_investments, df_scenario2_subsidies, df_scenario2_sum, "tab_scenario2")
+
+        # Tab-Steuerelemente
+        tabs_html = """<div class='tabs'>
+            <button class='tab-button' style="background-color: #1b9e77;" onclick="showTab('tab_scenario1', '#1b9e77')">Szenario 1</button>
+            <button class='tab-button' style="background-color: #7570b3;" onclick="showTab('tab_scenario2', '#7570b3')">Szenario 2</button>
+        </div>"""
+
+        # Zusammenfügen
+        full_html = f"""<style>
+            .tabs {{ margin-bottom: 0; position: relative; top: 0; left: 0; }}
+            .tab-button {{ margin-right: 5px; padding: 10px; color: white; border: none; cursor: pointer; }}
+            .tab-button:hover {{ opacity: 0.9; }}
+            .table {{ display: none; width: 100%; margin: 0; border-collapse: collapse; }}
+            .table th, .table td {{ border: 1px solid lightgrey; padding: 10px; text-align: left; }}
+            .table th {{ width: 70%; border-right: none; border-left: none; background-color: #f9f9f9; }}
+            .tab_scenario1 {{ display: table; border: 10px solid #1b9e77; border-radius: 10px; }}
+            .tab_scenario2 {{ border: 10px solid #7570b3; border-radius: 10px; }}
+        </style>
+        <script>
+            function showTab(tabName, borderColor) {{
+                var tables = document.querySelectorAll('.table');
+                tables.forEach(function(table) {{
+                    table.style.display = 'none';
+                    table.style.border = '10px solid ' + borderColor;
+                    table.style.borderRadius = '10px';
+                }});
+                document.querySelectorAll('.' + tabName).forEach(function(table) {{
+                    table.style.display = 'table';
+                }});
+            }}
+        </script>
```

Comment:
This should be moved into results.html or results.js using templating engine.
I could explain/help you there.